test/jasmine/tests/select_test.js

```diff
@@ -159,6 +165,42 @@ describe('select box and lasso', function() {
         });
     });
 
+    var selectingCnt, selectingData, selectedCnt, selectedData, deselectCnt, doubleClickData;
+    var selectedPromise, deselectPromise;
+
+    function resetEvents(gd) {
```

Comment:
Ha. This will conflict massively with https://github.com/plotly/plotly.js/pull/2030/commits/e53a908d725489c09857c64be00693fa6388d623 

Not sure who should get priority here.

Reply:
#2030 is epic enough already, you don't need to add a big merge resolution to it. You can merge first.